freqtrade/optimize/backtesting.py

```diff
@@ -354,38 +354,61 @@ def _get_ohlcv_as_lists(self, processed: Dict[str, DataFrame]) -> Dict[str, Tupl
 
     def _get_close_rate(self, sell_row: Tuple, trade: LocalTrade, sell: SellCheckTuple,
                         trade_dur: int) -> float:
+        leverage = trade.leverage or 1.0
+        is_short = trade.is_short or False
+        filled_dur = int((trade.close_date_utc - trade.buy_filled_date_utc).total_seconds() // 60)
```

Comment:
I'd kindly ask you to not mix these 2 things.

it's one thing to adjust the calculation for leverage.
but this change (filled_dur, the new column, ...) is something entirely different which does not belong into this PR in my opinion - which should aim to only fix the calculation for leverage.

Assuming you're interested in keeping this part as a separate PR - it's also done wrongly - you'll not need a new column on the trade object. The date is set on the order - so you can get it via `trade.orders[0].order_filled_date`.

Reply:
ok. I'll move out the fill_date stuff later

Reply:
I'm not saying it won't need fixing - but i'd rather not have it combined here - as it'll make the review more difficult (it'll mix 2 things which both affect calculation).

Reply:
I understand. Keeping it clean is what every coder likes.Although I'm not very good at this.😄